src/software/ai/hl/stp/stp.cpp

```diff
@@ -345,5 +369,13 @@ std::map<std::shared_ptr<const Tactic>, Robot> STP::assignRobotsToTactics(
         robots = remaining_robots;
     }
 
+    if (recalculate_tactic_assignment || (old_assignment > (new_assignment + NEW_TACTIC_ASSIGNMENT_THRESHOLD)))
+    {
+        robot_tactic_assignment = *new_tactic_assignment;
```

Comment:
I think raw pointers need to be avoided. I added a `delete new_tactic_assignment` here and it changed things (for the better maybe)

Reply:
no more raw pointers